File: Proyectos/ClubCiclistaJava/ElClub/src/el_club/Ciclista.java
package el_club;

// Ejemplo de acceso a los elementos de una clase desde métodos de instancia de otra clase. 
// Quita los comentarios  y prueba
public class Ciclista {
    
    private String nombre;
    private int edad;
    private Club miClub;    
    private Bicicleta miBicicleta;
    
    public Ciclista(String nom,int edad, Bicicleta bici, Club club){
        this.setNombre(nom);
        this.setEdad(edad);
        this.setBicicleta(bici);
        this.setClub(club);
    }
    
    public String getNombre(){
        return nombre;
    }
    
     public void setNombre(String nom){     
         nombre = nom;    
     }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public void setBicicleta(Bicicleta bici) {
       miBicicleta = bici;
    }
    public void setClub(Club club){
        this.miClub = club;
    }
    public void salirEnBici(double km){
      miBicicleta.incrementarKilometros(km);
    }
    public double cuantoskilometros(){
        return miBicicleta.getKilometros();    
    }
    public int ruedasDeMiBici(){
       return miBicicleta.NUMERORUEDAS;   
    }
    
    public String toString(){
        return "Me llamo " + nombre + " tengo " +edad+ " años " + 
                " pertenezco al club de ciclismo " + miClub.getNombre() + 
                " tengo una bici " + miBicicleta.toString();
    
    }
     
}

File: Proyectos/club_ciclista_ruby/el_club_ruby/lib/ciclista.rb
#encoding: UTF-8


require_relative 'bicicleta.rb'

module El_Club 
  class Ciclista
  
    attr_accessor :nombre, :edad, :miBicicleta, :miClub # se generan de forma automática los consultores y modificadores básicos
  
    def initialize(nom,edad, bici,club)
      @nombre=nom;
      @edad=edad;
      @miBicicleta=bici;
      @miClub = club
    end  
    
    def salirEnBici(km)
         miBicicleta.incrementarKilometros(km); # invocando a un método privado
    end
    
    def salirEnBiciHasta(lugar,km)
         miBicicleta.hacerRuta(lugar,km); 
    end
    
    def cuantoskilometros
        return miBicicleta.kilometrosRecorridos;    
    end
    
    def ruedasDeMiBici
        return Bicicleta.numeroRuedas; # invocando a un método de clase
    end
    
    def to_s
      "Me llamo #{@nombre} tengo #{@edad}  años  
        pertenezco al club de ciclismo #{@miClub.nombre} 
        tengo una bici, #{@miBicicleta.to_s}"
    end
end
end # fin de modulo UnModulo
 

